src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs

```diff
@@ -22,6 +22,7 @@ public class HttpRequestMessage : IDisposable
         private Uri? _requestUri;
         private HttpRequestHeaders? _headers;
         private Version _version;
+        private HttpVersionPolicy _versionPolicy;
```

Comment:
Should the `public enum HttpVersionPolicy` be defined as `: byte` then? Since it isn't atm.

Reply:
And by packed into a single int you mean encode those values into a single field (first byte `_sendStatus`, second byte `_versionPolicy`, third byte `_disposed`) or just putting them next to each other to improve layout of the type in memory?
If encoding into a single field, how do we then handle `Interlocked.CompareExchange` we currently do on `_sendStatus`? And is it worth the code complication?

Reply:
> Should the public enum HttpVersionPolicy be defined as : byte then? Since it isn't atm.

Nope, we typically don't do this unless an enum maps to some protocol.

> And by packed into a single int you mean encode those values into a single field (first byte _sendStatus, second byte _versionPolicy, third byte _disposed)

Yea, something like `_sendstatus | (versionPolicy << 8) | (_disposed << 16)` would be fine (or pack more tightly as none of the values need 8 bits, but that doesn't matter).

> If encoding into a single field, how do we then handle Interlocked.CompareExchange we currently do on _sendStatus? And is it worth the code complication?

Yea all access to the field would need to use `CompareExchange` and `Volatile.Read`, which is kind of unfortunate. Maybe the atomic one should live in its own variable to avoid this.

> is it worth the code complication?

We make a similar optimization elsewhere (on the response? I can't remember). It'll shave 8 bytes per request here.

Reply:
> on the response?
Nope, not there.

Reply:
I won't do the packing here, we can improve on that in the 6.0.